File: app/src/main/java/quintype/com/templatecollectionwithrx/models/collection/CoverImageMetadata.java
package quintype.com.templatecollectionwithrx.models.collection;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by akshaykoul on 18/04/16.
 */
public class CoverImageMetadata implements Parcelable {
    @SerializedName("width")
    @Expose
    private int width;
    @SerializedName("height")
    @Expose
    private int height;
    @SerializedName("mime-type")
    @Expose
    private String mimeType;

    public static final Creator<CoverImageMetadata> CREATOR = new Creator<CoverImageMetadata>() {
        @Override
        public CoverImageMetadata createFromParcel(Parcel in) {
            return new CoverImageMetadata(in);
        }

        @Override
        public CoverImageMetadata[] newArray(int size) {
            return new CoverImageMetadata[size];
        }
    };

    public int width() {
        return width;
    }

    public int height() {
        return height;
    }

    public String mimeType() {
        return mimeType;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.width);
        dest.writeInt(this.height);
        dest.writeString(this.mimeType);
    }

    public CoverImageMetadata() {
    }

    protected CoverImageMetadata(Parcel in) {
        this.width = in.readInt();
        this.height = in.readInt();
        this.mimeType = in.readString();
    }
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/collection/CoverImageMetadata.kt
package quintype.com.templatecollectionwithrx.models.collection

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 8/31/18.
 */

class CoverImageMetadata : Parcelable {
    @SerializedName("width")
    @Expose
    private val width: Int
    @SerializedName("height")
    @Expose
    private val height: Int
    @SerializedName("mime-type")
    @Expose
    private val mimeType: String

    fun width(): Int {
        return width
    }

    fun height(): Int {
        return height
    }

    fun mimeType(): String {
        return mimeType
    }


    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(this.width)
        dest.writeInt(this.height)
        dest.writeString(this.mimeType)
    }

    protected constructor(parcel: Parcel) {
        this.width = parcel.readInt()
        this.height = parcel.readInt()
        this.mimeType = parcel.readString()
    }

    companion object CREATOR : Parcelable.Creator<CoverImageMetadata> {
        override fun createFromParcel(parcel: Parcel): CoverImageMetadata {
            return CoverImageMetadata(parcel)
        }

        override fun newArray(size: Int): Array<CoverImageMetadata?> {
            return arrayOfNulls(size)
        }
    }

}
